File: 2441.c
int findMaxK(int* nums, int numsSize) {
    int best = -1;
    for (int i=0; i < numsSize; i++) {
        for (int j=i+1; j < numsSize; j++) {
            if (nums[j] == -1*nums[i]) {
                if (nums[j] > nums[i]) {
                    if (nums[j] > best) {
                        best = nums[j];
                    }
                }
                else {
                    if (nums[i] > best) {
                        best = nums[i];
                    }
                }
            }
        }
    }
    return best;
}

File: code/2441.py
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)
        best = -1
        neg = set()
        i = 0
        while i < n and nums[i] < 0:
            neg.add(nums[i]*-1)
            i += 1
        while i < n:
            if nums[i] in neg:
                best = nums[i]
            i += 1
        return best
